src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -83,9 +79,27 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
     @Override
     public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        MatchAllQueryBuilder query = new MatchAllQueryBuilder();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public Query toQuery(QueryParseContext parseContext) {
+        if (boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        //LUCENE 4 UPGRADE norms field is not supported anymore need to find another way or drop the functionality
+        //MatchAllDocsQuery query = new MatchAllDocsQuery(normsField);
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
```

Comment:
agreed on dropping, which we kinda did already :) how do proceed though? upstream? in our branch? how we do we track of these unrelated changes (I fear there's going to be many given that the deeper you dig the more you find...)?